src/main/cpp/tests/cast_string.cpp

```diff
@@ -473,3 +481,138 @@ TEST_F(StringToDecimalTests, Edges)
     CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
   }
 }
+
+TYPED_TEST(StringToFloatTests, Simple)
+{
+  cudf::test::strings_column_wrapper in{"-1.8946e-10",
+                                        "0001",
+                                        "0000.123",
+                                        "123",
+                                        "123.45",
+                                        "45.123",
+                                        "-45.123",
+                                        "0.45123",
+                                        "-0.45123",
+                                        "999999999999999999999",
+                                        "99999999999999999999",
+                                        "9999999999999999999",
+                                        "18446744073709551609",
+                                        "18446744073709551610",
+                                        "18446744073709551619999999999999",
+                                        "-18446744073709551609",
+                                        "-18446744073709551610",
+                                        "-184467440737095516199999999999997"};
+
+  std::vector<bool> valids{1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+  auto expected =
+    cudf::strings::to_floats(strings_column_view(in), cudf::data_type{type_to_id<TypeParam>()});
+  expected->set_null_mask(cudf::test::detail::make_null_mask(valids.begin(), valids.end()));
```

Comment:
This is less error-prone:
```suggestion
  auto valid_iter = cudf::test::iterators::no_nulls();
  auto expected =
    cudf::strings::to_floats(strings_column_view(in), cudf::data_type{type_to_id<TypeParam>()});
  expected->set_null_mask(cudf::test::detail::make_null_mask(valid_iter, valid_iter + expected->size()));
```